SecretSanta/SecretSanta.Business.Test/GlobalSuppressions.cs

```diff
@@ -0,0 +1,14 @@
+﻿// This file is used by Code Analysis to maintain SuppressMessage
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given
+// a specific target and scoped to a namespace, type, member, etc.
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.GiftTest.Gift_User_IsNull")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.GiftTest.Gift_URL_IsNull")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.GiftTest.Gift_Description_IsNull")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.GiftTest.Gift_Title_IsNull")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.GiftTest.Gift_WithAll_ProperInput")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.UserTest.User_GiftList_IsNUll")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.UserTest.USer_LastName_IsNUll")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.UserTest.USer_FirstName_IsNUll")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods", Scope = "member", Target = "~M:SecretSanta.Business.Test.UserTest.User_WithAll_ProperInput")]
```

Comment:
```suggestion
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "Under Score Not Allowed In Methods Does Not Apply In Test Methods")]
```
Remove Scope and Target to apply to all occurrences of the warning, that way you don't have multiple suppress messages for the same warning code. Also, you need to move this to the solution level and add to projects as a link, so all projects can reference the same suppression file. I recommend having a GlobalSuppressions file for global suppression and a TestSuppressions file for test specific suppression.

Reply:
Thanks